pkgs/tools/networking/dd-agent/integrations-core.nix

```diff
@@ -0,0 +1,66 @@
+{ pkgs
+, python
+, overrides ? (self: super: {})
+}:
+
+with pkgs.lib;
+
+let
+  src = pkgs.fetchFromGitHub {
+    owner = "DataDog";
+    repo = "integrations-core";
+    rev = "7be76e73969a8b9c993903681b300e1dd32f4b4d";
+    sha256 = "1qsqzm5iswgv9jrflh5mvbz9a7js7jf42cb28lzdzsp45iwfs2aa";
+  };
+  version = "git-2018-05-27";
+
+  buildIntegration = { pname, ... }@args: python.pkgs.buildPythonPackage (args // {
+    inherit src version;
+    name = "datadog-integration-${pname}-${version}";
+
+    postPatch = ''
+      # jailbreak install_requires
+      sed -i 's/==.*//' requirements.in
+      cp requirements.in requirements.txt
+    '';
+    sourceRoot = "source/${args.sourceRoot or pname}";
+    doCheck = false;
+  });
+
+  packages = (self: {
+    python = python.withPackages (ps: with self; [ disk network postgres nginx mongo ]);
+
+    datadog_checks_base = buildIntegration {
+      pname = "checks-base";
+      sourceRoot = "datadog_checks_base";
+      propagatedBuildInputs = with self; with python.pkgs; [ requests protobuf prometheus_client uuid simplejson uptime ];
+    };
+
+    disk = buildIntegration {
+      pname = "disk";
+      propagatedBuildInputs = with self; with python.pkgs; [ datadog_checks_base psutil ];
+    };
+
+    network = buildIntegration {
+      pname = "network";
+      propagatedBuildInputs = with self; with python.pkgs; [ datadog_checks_base psutil ];
+    };
+
+    postgres = buildIntegration {
+      pname = "postgres";
+      propagatedBuildInputs = with self; with python.pkgs; [ datadog_checks_base pg8000 psycopg2 ];
+    };
+
+    nginx = buildIntegration {
+      pname = "nginx";
+      propagatedBuildInputs = with self; with python.pkgs; [ datadog_checks_base ];
+    };
+
+    mongo = buildIntegration {
+      pname = "mongo";
+      propagatedBuildInputs = with self; with python.pkgs; [ datadog_checks_base pymongo ];
+    };
+
+  });
+
+in fix' (extends overrides packages)
```

Comment:
If I read this correctly the use of `fix'` is mostly to refer to `datadog_checks_base` recursively, right?

If I wanted to extend this with, say, the `ntp` integration, it isn't immediately obvious how I could reference `buildIntegration` when passing in `overrides`.

If there's a way to easily "enable" building of more integrations from the `integrations-core` repository I think it'd be great if this file had some comments explaining the reasoning and a usage example.

Maybe an alternative way is again to resort to a simple attribute set argument where `key :: IntegrationName`, where that `IntegrationName` is the `pname` of an integration in the `integrations-core` repository and `value:: [Package] -> [Package]`, which would practically be a filter for the integrations Python dependencies, for example:

```nix
pkgs.datadog-integrations-core {
  # core integrations built in addition to the defaults:
  coreIntegrations = {
    "ntp" = (pythonPkgs: []); # no extra dependencies
    "riak" = (pythonPkgs: [ pythonPkgs.httplib2 ]); # one extra dependency
  };

  # plus similar to the existing overrides arbitrary new integrations
  # (e.g. user-supplied ones):
  extraIntegrations = [ my-great-integration ];
};
```

Which could then be processed like so for the `coreIntegrations`: 

```nix
integrations = [ disk network postgres nginx mongo ] ++
  lib.mapAttrs (pname: fdeps: buildIntegration {
    inherit pname;
    propagatedBuildInputs = [ datadog_checks_base ] ++ (fdeps python.pkgs);
  }) cfg.coreIntegrations;
```

With `datadog_checks_base` declared in the `let`-binding outside of the returned attribute set (and maybe also inherited into it for convenience when building custom integrations).